### ISSUE TITLE
[Question/Suggestion] Possible to add dragons to custom datapack dimensions?

### ISSUE THREAD

**Issue by Xplodin** [Alex-the-666/Ice_and_Fire#5188]
The documentation describes how to add worldgen mod's biomes to iceandfire 
but it doesnt mention how to register lets say, a custom dimension accessible to ad astra thats a superflat world with plains as the biome.

here is what I added to dragontype_dragon_biomes.json (lightning was this case) inside:
[
	  {
        "type": "BIOME_TAG",
        "negate": false,
        "value": "minecraft:is_superflat"
      },
	  {
        "type": "BIOME_TAG",
        "negate": false,
        "value": "minecraft:plains"
      }
    ]

I then boot up the game, tpx to dimension, and dragons are nowhere to be found no matter the distance travelled.
Does Ice and Fire not work with superflat?
If it does how could I fix this to work in a datapack dimension with its own name?

Sorry for asking here, I tried asking this in a related issue but no response

**Comment by SiverDX** [comment]
is_superflat is not a biome tag thats a world preset
also i doubt ad astra uses plains biomes 
(https://github.com/terrarium-earth/Ad-Astra/tree/1.20.x/common/src/main/generated/resources/data/ad_astra/worldgen/biome)

you should look up what biome (tags) are

**Comment by Xplodin** [comment]
> is_superflat is not a biome tag thats a world preset also i doubt ad astra uses plains biomes (https://github.com/terrarium-earth/Ad-Astra/tree/1.20.x/common/src/main/generated/resources/data/ad_astra/worldgen/biome)
> 
> you should look up what biome (tags) are

I made a custom dimension for ad astra, via datapacks.
Im not really sure how id add that dimension I added to the mod to the config tbh

**Comment by SiverDX** [comment]
a dimension without biomes doesn't exist
add the biomes of your dimension to the config

**Comment by Xplodin** [comment]
I added plains to my superflat dimension.
It still refused to spawn dragons there despite having the plains biome tag

**Comment by SiverDX** [comment]
if it's an actual superflat world type then that might be the issue as that is known to not properly work with structures and features

(due to them requiring certain blocks to be replaced or starting at certain heights etc.)

**Comment by Xplodin** [comment]
yeah, I believe I only modified the normal biomes and not the cave and skeleton ones. Although am not sure if itll still work. Its a relatively small superflat layer wise. I could expand the layers and see

also does I&F use grass blocks to determine spawn?
and whats the height for roosts?

edit: I got other structures to spawn in the said dimensions. Like AE2 meteors, and graveyards from I&F
no luck on dragon roosts though

**Comment by OracleHisty** [comment]
> yeah, I believe I only modified the normal biomes and not the cave and skeleton ones. Although am not sure if itll still work. Its a relatively small superflat layer wise. I could expand the layers and see

_I'd suggest altering the base vanilla world gen format in your raw dimension to be a default world type WITH a surface rule. I'll look into a datapack template to fix that issue "globally" as a drop in fix for stuff like that._

Edit: Ultimately this is just a "Fake Superflat" solution.
